server/data/data-library/src/main/java/org/apache/james/rrt/lib/AbstractRecipientRewriteTable.java

```diff
@@ -205,37 +207,39 @@ public void removeRegexMapping(String user, Domain domain, String regex) throws
 
     @Override
     public void addAddressMapping(String user, Domain domain, String address) throws RecipientRewriteTableException {
-        String addressWithDomain = addDefaultDomainIfNone(address);
-        checkAddressIsValid(addressWithDomain);
-        checkMapping(user, domain, MappingImpl.address(addressWithDomain));
-        LOGGER.info("Add address mapping => {} for user: {} domain: {}", addressWithDomain, user, domain.name());
-        addMapping(user, domain, MappingImpl.address(addressWithDomain));
+        Mapping mapping = MappingImpl.address(address)
+            .appendDomainIfNone(defaultDomainSupplier());
+
+        checkHasValidAddress(mapping);
+        checkMapping(user, domain, mapping);
+
+        LOGGER.info("Add address mapping => {} for user: {} domain: {}", mapping, user, domain.name());
+        addMapping(user, domain, mapping);
     }
 
-    private String addDefaultDomainIfNone(String address) throws RecipientRewriteTableException {
-        if (address.indexOf('@') < 0) {
+    private Supplier<Domain> defaultDomainSupplier() throws RecipientRewriteTableException {
```

Comment:
we definitely don't need a method returning a Supplier that sneakyThrow an exception that the caller declare in its signature.
Please consider making appendDomainIfNone handle throwing methods to remove most of this boilerplate.

```
interface ThrowingDomainSupplier {
 Domain get() throw RecipientRewriteTableException;
}
Mapping appendDomainIfNone(ThrowingDomainSupplier supplier) throw RecipientRewriteTableException;
```